### PR TITLE
Refactor/router add switch

### PR DESCRIPTION
**Pull request by alexthompson207**
# PR PlanetParty
**Connor A-L, Alex T & Katie B**

### What does this do?

- [X] Fix
- [ ] Feature
- [X] Refactor
- [ ] Style
- [ ] Testing

### Are there any known bugs?

- [ ] No
- [X] Yes (if so please disclose in Notes for Reviewer & Next Iterations)
When a valid URL is entered for a single planet ex. /mars, the error message renders first before the single planet view. This is most likely because a fetch request is called when entering a URL /mars. This might be fixed with an added conditional render statement checking for state.error.

### Summary of the changes
1. Refactor `Router` using `Switch` components. Create a error message for when URL path doesn't match home or single planet view.
2. Move `<main>` tags to surround router, so each `Route` isn't wrapped in a `main` tag.

### What ticket(s) do the changes resolve?

#37 Router Refactor
Please Link Issues to this PR as necessary.

### Notes for the Reviewer
See Bug issue above ^

### How to Test Changes?
Pull down the branch, the site should work as before if paths names don't match home or single planet view an error should display

### Next Iterations

Add additional condition to render of single planet view to check for state error

### CODE REVIEW

**Review by ConnorAndersonLarson — APPROVED** [review]
Looks good! Thanks for going over this update with us in our meeting

File: src/components/App/App.tsx [hunk 1]
```diff
@@ -64,39 +64,30 @@ class App extends React.Component<{}, AllData> {
...
+                    {!this.state.error && !this.state.allPlanets.length && <h2>Loading...</h2>}
+                  <Planetarium allPlanets={this.state.allPlanets} sortKey={this.state.sortKey} error={this.state.error}/>
+                </>
+              )
+            }} />
+            <Route exact path="/:name" render={({ match }) => {
+              const foundPlanet = this.state.allPlanets.find(planet => planet.name.toLowerCase() === match.params.name.toLowerCase());
+              return (
+                <>
+                  {!foundPlanet && <h2>Oops, looks like that planet is out of our solar system</h2>}
```
Comment (knees4bees) [inline src/components/App/App.tsx:83]: Love this error message!

File: src/components/App/App.tsx [hunk 2]
```diff
@@ -64,39 +64,30 @@ class App extends React.Component<{}, AllData> {
...
+                  {foundPlanet && <PlanetInfo currentPlanet={foundPlanet} />}
+                </>
+              )
+            }} />
+            <Route path='*' render={() => <h2>Oops, looks like that planet is out of our solar system</h2>} />
```
Comment (knees4bees) [inline src/components/App/App.tsx:88]: Nice, simple solution to catching bad URLs

**Review by knees4bees — APPROVED** [review]
Great job, Alex! The code is much cleaner now.

**Pull request merged**